pyproject.toml

```diff
@@ -0,0 +1,46 @@
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
```

Comment:
This PR is growing large.  I'd rather do the Python 2.7 drop and the pyproject.toml conversion separately.

Reply:
Opened Python 2 drop #7, will open pyproject after it is merged since it reflects changes from it.